lib/matplotlib/colors.py

```diff
@@ -362,9 +362,14 @@ def to_rgba_array(c, alpha=None):
         pass
 
     if isinstance(c, str):
-        raise ValueError("Using a string of single character colors as "
-                         "a color sequence is not supported. The colors can "
-                         "be passed as an explicit list instead.")
+        if len(c):
+            raise ValueError("Using a string of single character colors as "
+                             "a color sequence is not supported. The colors "
+                             "can be passed as an explicit list instead.")
+        else:
+            raise ValueError("Using an empty string for color is not "
+                             "supported. Use 'none' if you do not want any "
+                             "color.")
```

Comment:
"string of single characters" was deprecated in 3.2 and removed in 3.4 (#18069). IMHO we don't need to bother users with that anymore - It's even more likely nowadays that they have just made a typo like 'yellwo'.

We should at least change the first message to `raise ValueError(f"{c!r} is not a valid color value")`

I even argue that the empty distinction is not necessary. We don't do this in other parts of the code either (c.f. `plt.plot([0], color='')`. I'd just go for

```suggestion
        raise ValueError(f"{c!r} is not a valid color value")
```